rotkehlchen/db/dbhandler.py

```diff
@@ -1075,16 +1074,24 @@ def add_blockchain_accounts(
             account_data: list[BlockchainAccountData],
     ) -> None:
         # Insert the blockchain account addresses and labels to the DB
-        tuples = [(
-            entry.chain.value,
-            entry.address,
-            entry.label,
-        ) for entry in account_data]
+        query_list = []
+        labels_exists = False
+        for entry in account_data:
+            query_list.append((entry.chain.value, entry.address))
+            if entry.label is not None:
+                labels_exists = True
         try:
             write_cursor.executemany(
-                'INSERT INTO blockchain_accounts(blockchain, account, label) VALUES (?, ?, ?)',
-                tuples,
+                'INSERT INTO blockchain_accounts(blockchain, account) VALUES (?, ?)', query_list,
             )
+            if labels_exists:
+                write_cursor.executemany(
+                    'INSERT OR REPLACE INTO address_book(address, blockchain, name) VALUES (?, ?, ?)',  # noqa: E501
+                    [
+                        (entry.address, entry.chain.value, entry.label)
+                        for entry in account_data if entry.label is not None
```

Comment:
you are doing this again man wtf. I told you. **Do not iterate the list two times**. You have already done this iteration at line 1079!!!!!!